File: CS/WebSolution2.Module.Web/MyFrameTemplateFactory.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.ExpressApp.Templates;
using DevExpress.ExpressApp;
using System.Web.Security;
using System.Web;
using DevExpress.ExpressApp.Web;
using System.Web.UI;

namespace WebSolution2.Module.Web {
    public class MyFrameTemplateFactory : IFrameTemplateFactory {
        private IFrameTemplate CreateFrameTemplateByVirtualPath(string path) {
            return (IFrameTemplate)System.Web.Compilation.BuildManager.CreateInstanceFromVirtualPath(path, typeof(Page));
        }
        #region IFrameTemplateFactory Members
        public IFrameTemplate CreateTemplate(TemplateContext context) {
            IFrameTemplate result = null;
            if (context == TemplateContext.LogonWindow) {
                result = CreateFrameTemplateByVirtualPath(FormsAuthentication.LoginUrl);
            } else if (context == TemplateContext.ApplicationWindow) {
                result = CreateFrameTemplateByVirtualPath(VirtualPathUtility.GetDirectory(FormsAuthentication.DefaultUrl) + UserDefaultTemplateProvider.CurrentUserDefaultTemplate);
            } else if (context == TemplateContext.PopupWindow) {
                result = CreateFrameTemplateByVirtualPath(VirtualPathUtility.GetDirectory(FormsAuthentication.DefaultUrl) + WebApplication.PopupWindowTemplatePage);
            } else if (context == TemplateContext.NestedFrame) {
                if (WebWindow.CurrentRequestPage == null) {
                    throw new Exception(string.Format(
                        "Cannot load the '{0}' control because the CurrentRequestPage is null",
                        WebApplication.NestedFrameTemplateControl));
                }
                Control nestedFrameTemplateControl = WebWindow.CurrentRequestPage.LoadControl(WebApplication.NestedFrameTemplateControl);
                nestedFrameTemplateControl.ID = "NestedFrameTemplateControl";
                result = (IFrameTemplate)nestedFrameTemplateControl;
            }
            return result;
        }
        #endregion

    }
}

File: VB/WebSolution2.Module.Web/MyFrameTemplateFactory.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.ExpressApp.Templates
Imports DevExpress.ExpressApp
Imports System.Web.Security
Imports System.Web
Imports DevExpress.ExpressApp.Web
Imports System.Web.UI

Namespace WebSolution2.Module.Web
	Public Class MyFrameTemplateFactory
		Implements IFrameTemplateFactory
		Private Function CreateFrameTemplateByVirtualPath(ByVal path As String) As IFrameTemplate
			Return CType(System.Web.Compilation.BuildManager.CreateInstanceFromVirtualPath(path, GetType(Page)), IFrameTemplate)
		End Function
		#Region "IFrameTemplateFactory Members"
        Public Function CreateTemplate(ByVal context As TemplateContext) As IFrameTemplate Implements IFrameTemplateFactory.CreateTemplate
            Dim result As IFrameTemplate = Nothing
            If context = TemplateContext.LogonWindow Then
                result = CreateFrameTemplateByVirtualPath(FormsAuthentication.LoginUrl)
            ElseIf context = TemplateContext.ApplicationWindow Then
                result = CreateFrameTemplateByVirtualPath(VirtualPathUtility.GetDirectory(FormsAuthentication.DefaultUrl) + UserDefaultTemplateProvider.CurrentUserDefaultTemplate)
            ElseIf context = TemplateContext.PopupWindow Then
                result = CreateFrameTemplateByVirtualPath(VirtualPathUtility.GetDirectory(FormsAuthentication.DefaultUrl) + WebApplication.PopupWindowTemplatePage)
            ElseIf context = TemplateContext.NestedFrame Then
                If WebWindow.CurrentRequestPage Is Nothing Then
                    Throw New Exception(String.Format("Cannot load the '{0}' control because the CurrentRequestPage is null", WebApplication.NestedFrameTemplateControl))
                End If
                Dim nestedFrameTemplateControl As Control = WebWindow.CurrentRequestPage.LoadControl(WebApplication.NestedFrameTemplateControl)
                nestedFrameTemplateControl.ID = "NestedFrameTemplateControl"
                result = CType(nestedFrameTemplateControl, IFrameTemplate)
            End If
            Return result
        End Function
		#End Region

	End Class
End Namespace